### PR TITLE
[17.0][MIG] sale_advance_payment: Migration to 17.0

### PR DESCRIPTION
**Pull request by Abranes**
Migration to 17.0

### CODE REVIEW

**Comment by Abranes** [comment]
I've reviewed the tests and it seems the error is because the currency conversion is not applied correctly. If anyone can provide some insight, otherwise I'll review it again when I get a chance.

**Comment by vvrossem** [comment]
Please Squash administrative commits with the previous commit for reducing commit noise. Check https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate for details.

File: sale_advance_payment/models/payment.py
Comment (vvrossem) [inline sale_advance_payment/models/payment.py:12]: Since Odoo 17, property `states` is no longer supported.
You should define the `readonly` attribute on the related views instead.
something of the like of `<field name="sale_id" readonly="state != 'draft'">`

File: sale_advance_payment/wizard/sale_advance_payment_wzd.py
Comment (vvrossem) [inline sale_advance_payment/wizard/sale_advance_payment_wzd.py:107]: https://github.com/odoo/odoo/blob/17.0/addons/sale/models/sale_order.py#L392-L401
The pricelist is not set anymore on draft SO. 
The Travis issue might come from this. 
I've ran the tests on the module in its version 16.0, and the currency_id doesn't match with the one of 17.0

**Review by vvrossem — CHANGES_REQUESTED** [review]

**Comment by vvrossem** [comment]
Have a look at [[17.0][MIG] purchase advance payment](https://github.com/OCA/purchase-workflow/pull/2234), it might be useful on what changes to do on the unit tests

**Comment by Abranes** [comment]
Thank you for the comments @vvrossem. I'm a bit busy right now, but I will review everything and make the necessary changes by the end of the week.

**Review by yankinmax — APPROVED** [review]

**Comment by rousseldenis** [comment]
@Abranes

File: pandoc-3.2-1-amd64.deb
Comment (Camille0907) [inline pandoc-3.2-1-amd64.deb:1]: This file should be removed, cf. https://github.com/OCA/oca-addons-repo-template/issues/262

File: sale_advance_payment/views/sale_view.xml
Comment (Camille0907) [inline sale_advance_payment/views/sale_view.xml:14]: We don't want it anymore to be invisible when the state is done ? It does not look like a migration action

**Review by Camille0907 — CHANGES_REQUESTED** [review]
Please remove the `.deb` file. Except that, migration looks technically good to me

**Review by gurneyalex — CHANGES_REQUESTED** [review]
Don't merge the PR while the debian file is in there.

**Comment by yankinmax** [comment]
This PR will be superseded by:
- [ ] https://github.com/OCA/sale-workflow/pull/3244

**Comment by gurneyalex** [comment]
superseeded by https://github.com/OCA/sale-workflow/pull/3244

**Pull request closed without merging**
